GocadRelatedScripts/Face.py

```diff
@@ -0,0 +1,188 @@
+
+    @classmethod
+    def from_ts(self, fid):
+        surface_name = "undefined"
+        xyzl = []
+        vid = {}
+        trl = []
+        prev_vert = -1
+        ivertex = 0
+        line = fid.readline()
+        if not line:
+            return None
+        title = line.split()
+        # Skip everything not Surface
+        if title[1].lower() != "tsurf":
+            print("skipping %s" % (title[1]))
+            while True:
+                line = fid.readline()
+                if not line:
+                    break
+                if line.startswith("END_"):
+                    continue
+                elif line.startswith("END"):
+                    break
+        else:
+            while True:
+                line = fid.readline()
+                if not line:
+                    break
+                if line.startswith("VRTX"):
+                    val = [float(val) for val in line.split()[1:5]]
+                    newVid = int(val[0])
+                    vid[newVid] = ivertex
+                    ivertex = ivertex + 1
+                    xyzl.append(val[1:4])
+                elif line.startswith("TRGL"):
+                    val = [int(val) for val in line.split()[1:4]]
+                    trl.append(val)
+                elif line.startswith("END_"):
+                    continue
+                elif line.startswith("name"):
+                    surface_name = (line.split(":")[1]).strip()
+                    print("now processing %s" % surface_name)
+                elif line.startswith("ATOM"):
+                    val = [int(val) for val in line.split()[1:3]]
+                    vid0 = vid[val[1]]
+                    xyzl.append(xyzl[vid0])
+                    vid[val[0]] = ivertex
+                    ivertex = ivertex + 1
+                elif line.startswith("END"):
+                    break
+            vertex = np.asarray(xyzl)
+            connect = np.asarray(trl)
+            myFace = Face(vertex, connect)
+            myFace.reindex(vid)
+            return myFace
+
+    def setup_proj_objects(self, sProj):
+        import pyproj
+
+        lla = pyproj.Proj(proj="latlong", ellps="WGS84", datum="WGS84")
+        if sProj != "geocent":
+            myproj = pyproj.Proj(sProj)
+        else:
+            myproj = pyproj.Proj(proj="geocent", ellps="WGS84", datum="WGS84")
+        return lla, myproj
+
+    def proj(self, sProj):
+        "project the node coordinate array"
+        import pyproj
+
+        lla, myproj = self.setup_proj_objects(sProj)
+        print("projecting the node coordinates")
+        self.vertex[:, 0], self.vertex[:, 1], self.vertex[:, 2] = pyproj.transform(lla, myproj, self.vertex[:, 0], self.vertex[:, 1], self.vertex[:, 2], radians=False)
+        print(self.vertex)
+        print("done projecting")
+
+    def convert_projected_to_latlon(self, sProj):
+        "Convert the already projected node coordinate array to lat/lon"
```

Comment:
Still, these two functions do essentially the same, but for the printed messages. If you do a projection or conversion depends on the content of `sProj`, right?

Reply:
right, lla and myproj should be switched!